src/events.rs

```diff
@@ -26,13 +37,29 @@ impl Default for Events {
 }
 
 impl Events {
+    /// Creates new event channel.
     pub fn new() -> Self {
         let (sender, receiver) = channel::bounded(1_000);
 
-        Self { receiver, sender }
+        Self {
+            receiver,
+            sender,
+            last_error: Arc::new(RwLock::new("".to_string())),
+        }
+    }
+
+    /// Set last error string.
+    /// Implemented as blocking as used from macros in different, not always async blocks.
```

Comment:
I think it's not a problem if we call `to_string()` before taking a lock. Or does Rust optimise this itself?